### PR TITLE
PP-888/contract-abstraction

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- As a developer, I want to remove the code duplication, so that the repo will be easier to maintain.

## Why

- Duplicated code

## Refs

-  [PP-888](https://rsklabs.atlassian.net/browse/PP-888)
-  [PP-889](https://rsklabs.atlassian.net/browse/PP-889)
-  [PP-890](https://rsklabs.atlassian.net/browse/PP-890)

### CODE REVIEW

File: contracts/interfaces/ISmartWalletRelayer.sol
Comment (antomor) [inline contracts/interfaces/ISmartWalletRelayer.sol:1]: I'd rename it to something like `IRelayedSmartWalletFactory`

**Review by antomor — CHANGES_REQUESTED** [review]
@franciscotobar Thank you for these changes. I've left some comments, but before approving them, I'd like to test them on our integration tests.

File: .solhint.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
             {
                 "ignoreConstructors": true
             }
-        ]
+        ],
+        "no-global-import": "off"
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply .solhint.json:31]: When you run the `npm run lint`, there were a lot of global import of path warnings, that were affecting the visibility from other error or warnings, we may fix this in another task or just turn it off.

File: contracts/factory/BaseSmartWalletFactory.sol [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+41 | 57    | JUMPI          | [0, rds]                                    | if(isSuccess) then jump to PC=43
+42 | FD    | REVERT         | []                                          | revert(Mem[0, rds-1])
+43 | 5B    | JUMPDEST       | [0, rds]                                    |
+44 | F3    | RETURN         | []                                          | return(Mem[0, rds-1])
+ */
+
+/** Factory of Proxies to the SmartWallet (Forwarder)
+ */
+abstract contract BaseSmartWalletFactory is
+    IWalletFactory,
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply contracts/factory/BaseSmartWalletFactory.sol:76]: Updated

File: contracts/interfaces/ISmartWalletRelayer.sol
Comment (franciscotobar) [reply contracts/interfaces/ISmartWalletRelayer.sol:1]: Updated

**Pull request merged**